### PR TITLE
Add Nonlinear Manifold Decoders for Operator Learning (NOMAD)

### PR DESCRIPTION
**Pull request by ven-k**
...as defined by https://arxiv.org/abs/2206.03551

- Add a test for Burger dataset.
  Returns 2.33 mean-diff (vs DeepONet returns 2.66)

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@ven-k can you enable buildkite CI here so it tests the GPU on this?

**Comment by ChrisRackauckas** [comment]
This is missing an add to the docs

**Comment by ven-k** [comment]
Ok. Will add them both

**Comment by ven-k** [comment]
Is there anything I should add to this?

File: docs/src/introduction.md [hunk 1]
```diff
@@ -24,3 +24,8 @@ by linking the operators into a Markov chain.
 Deep operator network (DeepONet) learns a neural operator with the help of two sub-neural network structures described as the branch and the trunk network.
 The branch network is fed the initial conditions data, whereas the trunk is fed with the locations where the target(output) is evaluated from the corresponding initial conditions.
 It is important that the output size of the branch and trunk subnets is same so that a dot product can be performed between them.
+
+## [Nonlinear Manifold Decoders for Operator Learning](https://github.com/SciML/NeuralOperators.jl/blob/master/src/NOMAD.jl)
+
+Nonlinear Manifold Decoders for Operator Learning (NOMAD) learns a neural operator with a nonlinear decoder parameterized by a deep neural network which jointly takes output of approximator and the locations as parameters.
+The approximator network is fed with the initial conditions data. The output-of-approximator and the locations are then passed to a decoder neural network to get the target (output). It is important that the input size of the decoder subnet is sum of size of the output-of-approximator and number of locations.
```
Comment (ChrisRackauckas) [inline docs/src/introduction.md:31]: It would be nice to enforce that constraint, but I don't see how to do it in general.

**Comment by ChrisRackauckas** [comment]
It looks good to me. I'll merge for now, but I really wonder if there's an easier way to support this by doing like, DeepONets have a reducer function which defaults to `sum` and then it can be a neural network or something. But ehh, that might just make it more complicated.

**Pull request merged**